email-auth-provider/src/handlers/set_password.rs

```diff
@@ -0,0 +1,121 @@
+#[ft_sdk::form]
+pub fn set_password(
+    mut conn: ft_sdk::Connection,
+    ft_sdk::Required(new_password): ft_sdk::Required<"new-password">,
+    ft_sdk::Required(new_password2): ft_sdk::Required<"new-password2">,
+    ft_sdk::Query(code): ft_sdk::Query<"code">,
+    ft_sdk::Query(spr): ft_sdk::Query<"spr">,
+    ft_sdk::Query(email): ft_sdk::Query<"email">,
+    ft_sdk::Query(next): ft_sdk::Query<"next", Option<String>>,
+    host: ft_sdk::Host,
+) -> ft_sdk::form::Result {
+    validate_email_and_password(&email, &new_password, &new_password2)?;
+
+    let next = next.unwrap_or_else(|| "/".to_string());
+
+    let (user_id, data) = match ft_sdk::auth::provider::user_data_by_custom_attribute(
+        &mut conn,
+        email_auth::PROVIDER_ID,
+        email_auth::PASSWORD_RESET_CODE_KEY,
+        &code,
+    ) {
```

Comment:
This means we will have to index the reset code key field also. I think we can search by email as well, and email index will def exist, so we can save one index by changing query.

Reply:
We currently do not have indexes, which we will fix someday.

Reply:
We've similar code in one `conrfirm_email`. I'll leave this one as it is and we can come back to it when we've added an index on the email field.